From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Vainomaa <mikroskeem@mikroskeem.eu>
Date: Thu, 23 Apr 2020 04:55:03 +0300
Subject: [PATCH] Toothpick specific API


diff --git a/src/main/java/eu/mikroskeem/toothpick/api/Toothpick.java b/src/main/java/eu/mikroskeem/toothpick/api/Toothpick.java
new file mode 100644
index 0000000000000000000000000000000000000000..6799f4426d522668aa514cd74cb4d34e275852e5
--- /dev/null
+++ b/src/main/java/eu/mikroskeem/toothpick/api/Toothpick.java
@@ -0,0 +1,14 @@
+package eu.mikroskeem.toothpick.api;
+
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Toothpick specific API
+ */
+public interface Toothpick {
+    /**
+     * Static instance of {@link Toothpick}
+     */
+    @NotNull
+    Toothpick INSTANCE = null;
+}
