From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Vainomaa <mikroskeem@mikroskeem.eu>
Date: Fri, 24 Apr 2020 00:05:19 +0300
Subject: [PATCH] Rip out legacy support


diff --git a/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java b/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
index 3a02dbe9d183bc907dcce081d8338d5716ed5242..bc98cfa094cb9b5b6dbec32048257c3299283707 100644
--- a/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
+++ b/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
@@ -141,7 +141,7 @@ public final class PluginClassLoader extends URLClassLoader { // Spigot
                         throw new ClassNotFoundException(name, ex);
                     }
 
-                    classBytes = loader.server.getUnsafe().processClass(description, path, classBytes);
+                    //classBytes = loader.server.getUnsafe().processClass(description, path, classBytes); // Toothpick - Rip out legacy support
 
                     int dot = name.lastIndexOf('.');
                     if (dot != -1) {
