From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ysl3000 <yannicklamprecht@live.de>
Date: Sat, 13 Jun 2020 15:46:55 +0200
Subject: [PATCH] add trigger block entity to api


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index 75c9225bd07b4c4b6dcd702c52294a7637eb6884..245c389b15514eea8a66ae5ed18ea810ee3a274d 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -45,6 +45,8 @@ import org.jetbrains.annotations.Nullable;
  */
 public interface World extends PluginMessageRecipient, Metadatable {
 
+    void triggerBlockEntityEvent(Location location, int paramA, int paramB);
+
     // Paper start
     /**
      * @return The amount of Entities in this world
