From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ysl3000 <yannicklamprecht@live.de>
Date: Sat, 13 Jun 2020 15:47:52 +0200
Subject: [PATCH] add method to allow enchantment registry again


diff --git a/src/main/java/org/bukkit/enchantments/Enchantment.java b/src/main/java/org/bukkit/enchantments/Enchantment.java
index b833ef63fbe01271ceb2bd83a9eb4a84c9912761..b8f6e8da6863f3aef4f76a144870804949defa40 100644
--- a/src/main/java/org/bukkit/enchantments/Enchantment.java
+++ b/src/main/java/org/bukkit/enchantments/Enchantment.java
@@ -354,6 +354,13 @@ public abstract class Enchantment implements Keyed {
         acceptingNew = false;
     }
 
+    /**
+     * Starts accepting any enchantment registrations
+     */
+    public static void startAcceptingRegistrations() {
+        acceptingNew = true;
+    }
+
     /**
      * Gets the Enchantment at the specified key
      *
