From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ysl3000 <yannicklamprecht@live.de>
Date: Sat, 13 Jun 2020 15:46:55 +0200
Subject: [PATCH] add trigger block entity to api


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index 9518da825ed752c5a477ca9132de50f923f9192d..5bdaaa42c2c39090c832b53fc4b3373c9fd5fead 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -45,6 +45,8 @@ import org.jetbrains.annotations.Nullable;
  */
 public interface World extends PluginMessageRecipient, Metadatable {
 
+    void triggerBlockEntityEvent(Location location, int paramA, int paramB);
+
     // Paper start
     /**
      * @return The amount of Entities in this world
