From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Vainomaa <mikroskeem@mikroskeem.eu>
Date: Thu, 23 Apr 2020 05:18:07 +0300
Subject: [PATCH] Brigadier command sender


diff --git a/src/main/java/net/minecraft/server/CommandListenerWrapper.java b/src/main/java/net/minecraft/server/CommandListenerWrapper.java
index 0b23a0548d52a30c064d624e39a896a9791aab3b..58973765d1a3d379d18be34e6989ab440cd9a96f 100644
--- a/src/main/java/net/minecraft/server/CommandListenerWrapper.java
+++ b/src/main/java/net/minecraft/server/CommandListenerWrapper.java
@@ -15,7 +15,7 @@ import java.util.function.BinaryOperator;
 import java.util.stream.Stream;
 import javax.annotation.Nullable;
 
-public class CommandListenerWrapper implements ICompletionProvider {
+public class CommandListenerWrapper implements ICompletionProvider, eu.mikroskeem.toothpick.api.sender.BrigadierCommandSender {
 
     public static final SimpleCommandExceptionType a = new SimpleCommandExceptionType(new ChatMessage("permissions.requires.player", new Object[0]));
     public static final SimpleCommandExceptionType b = new SimpleCommandExceptionType(new ChatMessage("permissions.requires.entity", new Object[0]));
@@ -54,14 +54,17 @@ public class CommandListenerWrapper implements ICompletionProvider {
         this.n = vec2f;
     }
 
+    @Override public CommandListenerWrapper withEntity(org.bukkit.entity.Entity entity) { return a(((org.bukkit.craftbukkit.entity.CraftEntity) entity).getHandle()); } // Toothpick - Brigadier command sender
     public CommandListenerWrapper a(Entity entity) {
         return this.k == entity ? this : new CommandListenerWrapper(this.base, this.d, this.n, this.e, this.f, entity.getDisplayName().getString(), entity.getScoreboardDisplayName(), this.i, entity, this.j, this.l, this.m);
     }
 
+    @Override public CommandListenerWrapper withPosition(org.bukkit.util.Vector vector) { return a(org.bukkit.craftbukkit.util.CraftVector.toNMS(vector)); } // Toothpick - Brigadier command sender
     public CommandListenerWrapper a(Vec3D vec3d) {
         return this.d.equals(vec3d) ? this : new CommandListenerWrapper(this.base, vec3d, this.n, this.e, this.f, this.g, this.h, this.i, this.k, this.j, this.l, this.m);
     }
 
+    @Override public CommandListenerWrapper withRotation(float yaw, float pitch) { return a(new Vec2F(yaw, pitch)); } // Toothpick - Brigadier command sender
     public CommandListenerWrapper a(Vec2F vec2f) {
         return this.n.c(vec2f) ? this : new CommandListenerWrapper(this.base, this.d, vec2f, this.e, this.f, this.g, this.h, this.i, this.k, this.j, this.l, this.m);
     }
@@ -100,6 +103,17 @@ public class CommandListenerWrapper implements ICompletionProvider {
         return this.b(argumentanchor_anchor.a(entity));
     }
 
+    // Toothpick start - Brigadier command sender
+    @Override
+    public CommandListenerWrapper withLookingAt(org.bukkit.util.Vector vector) {
+        try {
+            return b(org.bukkit.craftbukkit.util.CraftVector.toNMS(vector));
+        } catch (CommandSyntaxException e) {
+            com.destroystokyo.paper.util.SneakyThrow.sneaky(e);
+            throw null;
+        }
+    }
+    // Toothpick end
     public CommandListenerWrapper b(Vec3D vec3d) throws CommandSyntaxException {
         Vec3D vec3d1 = this.m.a(this);
         double d0 = vec3d.x - vec3d1.x;
@@ -138,19 +152,23 @@ public class CommandListenerWrapper implements ICompletionProvider {
     }
     // CraftBukkit end
 
+    @Override public org.bukkit.util.Vector getCurrentPosition() { return org.bukkit.craftbukkit.util.CraftVector.toBukkit(getPosition()); } // Toothpick - Brigadier command sender; also dumb naming conflict
     public Vec3D getPosition() {
         return this.d;
     }
 
+    @Override public org.bukkit.World getBukkitWorld() { return getWorld().getWorld(); } // Toothpick - Brigadier command sender
     public WorldServer getWorld() {
         return this.e;
     }
 
+    @Override public org.bukkit.entity.Entity getBukkitEntity() { var en = getEntity(); return en != null ? en.getBukkitEntity() : null; } // Toothpick - Brigadier command sender
     @Nullable
     public Entity getEntity() {
         return this.k;
     }
 
+    @Override public org.bukkit.entity.Entity getEntityOrThrow() throws Exception { return g().getBukkitEntity(); } // Toothpick - Brigadier command sender
     public Entity g() throws CommandSyntaxException {
         if (this.k == null) {
             throw CommandListenerWrapper.b.create();
@@ -159,6 +177,7 @@ public class CommandListenerWrapper implements ICompletionProvider {
         }
     }
 
+    @Override public org.bukkit.entity.Player getPlayerOrThrow() throws Exception { return h().getBukkitEntity(); } // Toothpick - Brigadier command sender
     public EntityPlayer h() throws CommandSyntaxException {
         if (!(this.k instanceof EntityPlayer)) {
             throw CommandListenerWrapper.a.create();
@@ -171,6 +190,7 @@ public class CommandListenerWrapper implements ICompletionProvider {
         return this.n;
     }
 
+    @Override public org.bukkit.Server getBukkitServer() { return getServer().server; } // Toothpick - Brigadier command sender
     public MinecraftServer getServer() {
         return this.i;
     }
