From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Vainomaa <mikroskeem@mikroskeem.eu>
Date: Thu, 23 Apr 2020 05:26:32 +0300
Subject: [PATCH] Allow overriding Mojang API url via system property


diff --git a/src/main/java/com/mojang/authlib/yggdrasil/YggdrasilGameProfileRepository.java b/src/main/java/com/mojang/authlib/yggdrasil/YggdrasilGameProfileRepository.java
index 23f1447cfcef89683a9bc98f0f912d76a1f93012..37497d242db924335c5f4abe689fea6262580868 100644
--- a/src/main/java/com/mojang/authlib/yggdrasil/YggdrasilGameProfileRepository.java
+++ b/src/main/java/com/mojang/authlib/yggdrasil/YggdrasilGameProfileRepository.java
@@ -18,7 +18,7 @@ import java.util.Set;
 
 public class YggdrasilGameProfileRepository implements GameProfileRepository {
     private static final Logger LOGGER = LogManager.getLogger();
-    private static final String BASE_URL = "https://api.mojang.com/";
+    private static final String BASE_URL = System.getProperty("toothpick.mojangapi.url", "https://api.mojang.com/");
     private static final String SEARCH_PAGE_URL = BASE_URL + "profiles/";
     private static final int ENTRIES_PER_PAGE = 2;
     private static final int MAX_FAIL_COUNT = 3;
