From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MiniDigger <admin@minidigger.me>
Date: Tue, 5 May 2020 23:42:19 +0200
Subject: [PATCH] Allow Java 15


diff --git a/src/main/java/org/bukkit/craftbukkit/Main.java b/src/main/java/org/bukkit/craftbukkit/Main.java
index cb60310e63ce0f55f5e706edf9e1ef61a3732600..d92c5069504ad4f47ac279f3b0bb1d76892efad5 100644
--- a/src/main/java/org/bukkit/craftbukkit/Main.java
+++ b/src/main/java/org/bukkit/craftbukkit/Main.java
@@ -173,8 +173,8 @@ public class Main {
             }
 
             float javaVersion = Float.parseFloat(System.getProperty("java.class.version"));
-            if (javaVersion > 58.0) {
-                System.err.println("Unsupported Java detected (" + javaVersion + "). Only up to Java 14 is supported.");
+            if (javaVersion > 59.0) { // MiniPaper - java 15!
+                System.err.println("Unsupported Java detected (" + javaVersion + "). Only up to Java 15 is supported."); // MiniPaper - java 15!
                 if (!Boolean.getBoolean("Paper.IgnoreJavaVersion")) return; // Paper
             }
 
