From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MiniDigger <admin@minidigger.me>
Date: Wed, 6 May 2020 17:07:40 +0200
Subject: [PATCH] Preremapping


diff --git a/src/main/java/com/destroystokyo/paper/util/misc/AreaMap.java b/src/main/java/com/destroystokyo/paper/util/misc/AreaMap.java
index f625da9f0968b6f41e302b628c4439198eb4ed64..f9c0f52920d27fb71ef4b252b3c9152dfc5a9a18 100644
--- a/src/main/java/com/destroystokyo/paper/util/misc/AreaMap.java
+++ b/src/main/java/com/destroystokyo/paper/util/misc/AreaMap.java
@@ -3,6 +3,7 @@ package com.destroystokyo.paper.util.misc;
 import com.destroystokyo.paper.util.math.IntegerUtil;
 import it.unimi.dsi.fastutil.longs.Long2ObjectLinkedOpenHashMap;
 import it.unimi.dsi.fastutil.longs.Long2ObjectOpenHashMap;
+import it.unimi.dsi.fastutil.longs.Long2ObjectMap;
 import it.unimi.dsi.fastutil.objects.Object2IntOpenHashMap;
 import it.unimi.dsi.fastutil.objects.Object2LongOpenHashMap;
 import net.minecraft.server.ChunkCoordIntPair;
@@ -134,10 +135,10 @@ public abstract class AreaMap<E> {
         final int centerX = MCUtil.getCoordinateX(currPosition);
         final int centerZ = MCUtil.getCoordinateZ(currPosition);

-        for (Iterator<Long2ObjectLinkedOpenHashMap.Entry<PooledLinkedHashSets.PooledObjectLinkedOpenHashSet<E>>> iterator = this.areaMap.long2ObjectEntrySet().fastIterator();
+        for (Iterator<Long2ObjectMap.Entry<PooledLinkedHashSets.PooledObjectLinkedOpenHashSet<E>>> iterator = this.areaMap.long2ObjectEntrySet().fastIterator();
              iterator.hasNext();) {

-            final Long2ObjectLinkedOpenHashMap.Entry<PooledLinkedHashSets.PooledObjectLinkedOpenHashSet<E>> entry = iterator.next();
+            final Long2ObjectMap.Entry<PooledLinkedHashSets.PooledObjectLinkedOpenHashSet<E>> entry = iterator.next();
             final long key = entry.getLongKey();
             final PooledLinkedHashSets.PooledObjectLinkedOpenHashSet<E> map = entry.getValue();

